include/networkit/graph/DFS.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef GRAPH_DFS_H
+#define GRAPH_DFS_H
+
+#include <queue>
+
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+namespace GraphTraversal {
+
+/**
+ * Iterate over nodes in depth-first search order starting from the given source node.
+ *
+ * @param G The input graph.
+ * @param source The source node.
+ * @param handle Takes a node as input parameter.
+ */
+template <typename L>
+void DFSfrom(const Graph &G, node source, L handle) {
```

Comment:
If we keep the functionality as is (see below), this is only a special case of `DFSEdgesFrom` and we should remove one implementation.

However, it's somewhat strange that BFS supports to report the depth of a node (yes, I know it's unweighted SSSP), while DFS does not. I cannot think of an application right now, but it seems strange to have different callback signatures for these dual algorithms; esp. when giving the depth comes basically for free.

Reply:
That's true, we might want to refactor this in a separate PR, the current one is already quite big.